File: leetcode/100-999/169. Majority Element/solution_1.cs
public class Solution 
{
    public int MajorityElement(int[] nums) 
    {
        int majority = -1;
        Dictionary<int, int> numAndCount = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; ++i)
        {
            if (numAndCount.ContainsKey(nums[i]))
                numAndCount[nums[i]]++;
            else
                numAndCount[nums[i]] = 1;

            if (numAndCount[nums[i]] > nums.Length / 2)
            {
                majority = numsAndCount[nums[i]];
                break;
            }
        }

        return majority;
    }
}

File: leetcode/100-999/169. Majority Element/solution_1.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var majorityElement = function(nums) 
{
    let majority = -1
    const numAndCount = new Set()

    for (let i = 0; i < nums.length; i++) 
    {
        if (numAndCount[nums[i]])
            numAndCount[nums[i]]++
        else 
            numAndCount[nums[i]] = 1

        if (numAndCount[nums[i]] > nums.length / 2)
        {
            majority = nums[i];
            break;
        }
    }

    return majority
};